src/main/scala/nubank/parsinho/parser/ExpressionDataParser.scala

```diff
@@ -0,0 +1,21 @@
+package nubank.parsinho.parser
+
+import nubank.parsinho.expressions._
+import nubank.parsinho.utils.UniqueAttributeId
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.types._
+
+object ExpressionDataParser {
+  @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
+  def fromExpression(expr: Expression): ExpressionData = {
+    println(expr, expr.getClass.getSimpleName)
+    expr match {
+      case a: Alias => AliasData(a.name, UniqueAttributeId.fromExprId(a.exprId), fromExpression(a.child))
+      case ar: AttributeReference => AttributeReferenceData(ar.name,ar.dataType,UniqueAttributeId.fromExprId(ar.exprId))
+      case l: Literal => LiteralData(l.toString(), l.dataType)
+
+      // Catch-all for unimplemented expressions
+      case ouch => NoExpressionData(ouch.getClass.getSimpleName, dataType = NullType, children = ouch.children.map(fromExpression))
```

Comment:
Just to match the Node implementation, WDYT about renaming this?
```suggestion
      case ouch => NonParsedExpression(ouch.getClass.getSimpleName, dataType = NullType, children = ouch.children.map(fromExpression))
```

Reply:
Perfect. Will do.